src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -122,24 +123,27 @@ const char* GetCalendarName(CalendarId calendarId)
 		case JAPANESELUNISOLAR:
 		case TAIWANLUNISOLAR:
 			return "chinese";
```

Comment:
I am not expecting JapaneseLunisolar/ChineseLuniSolar/TaiwanLunisolarCalendar/KoreanLunisolarCalendar/JulianCalendar be passed here at all because these are not associated with any culture and it is not defined in Linux (from the list you sent).
I would say use Taiwanese calendar with TW and not to use Chinese calendar in any other place till we support it (if we need to). 
Also it will be a good idea to know what Windows does in this issue too. I mean what calendars they use with CN, CX, HK, MO, SG. it should be easy to write some small code and run it on Windows 10 which create cultures for such regions and then enumerate the culture optional calendars. I can help in that if you like.